### PR TITLE
Remove the use of JsonRpcParameters from RpcMethods

### PR DESCRIPTION
**Pull request by rain-on**
The JsonRpcParameters are now "hidden" behind the JsonRpcRequest
object.

Signed-off-by: Trent Mohay <trent.mohay@consensys.net>

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/JsonRpcRequest.java [hunk 1]
```diff
@@ -124,4 +128,12 @@ public boolean equals(final Object o) {
   public int hashCode() {
     return Objects.hash(id, method, Arrays.hashCode(params), version, isNotification);
   }
+
+  public <T> T getRequiredParameter(final int index, final Class<T> paramClass) {
+    return parameterAccessor.required(params, index, paramClass);
+  }
+
+  public <T> Optional<T> getOptionaParameter(final int index, final Class<T> paramClass) {
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/JsonRpcRequest.java]:
```suggestion
  public <T> Optional<T> getOptionalParameter(final int index, final Class<T> paramClass) {
```

Comment (rain-on) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/JsonRpcRequest.java]: done

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/JsonRpcRequest.java [hunk 2]
```diff
@@ -31,6 +33,8 @@
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class JsonRpcRequest {
```
Comment (jframe) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/JsonRpcRequest.java]: Can the getParams be removed now that all access is done through the getRequiredParameter and getOptionalParameter methods

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/JsonRpcRequest.java [hunk 3]
```diff
@@ -31,6 +33,8 @@
...
 
+  private static final JsonRpcParameter parameterAccessor = new JsonRpcParameter();
```
[the remark this answers is not in the document]
Comment (rain-on) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/JsonRpcRequest.java]: done.

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/JsonRpcRequest.java (hunk 2 above)
Comment (rain-on) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/JsonRpcRequest.java]: Not quite - it's still used by a few RPCs - of which I'm not confident to change.

**Pull request merged**